salt/modules/lxc.py

```diff
@@ -800,6 +835,14 @@ def _network_conf(conf_tuples=None, **kwargs):
     for val in new.values():
         for row in val:
             ret.append(salt.utils.odict.OrderedDict([(row, val[row])]))
+    # on old versions of lxc, still support the gateway auto mode
+    # if we didnt explicitly say no to
+    # (lxc.network.ipv4.gateway: auto)
+    if (
+        distutils.version.LooseVersion(version()) <= '1.0.7' and
+        True not in ['ipv4.gateway' in a[0] for a in ret]
+    ):
+        ret.append({'lxc.network.ipv4.gateway': 'auto'})
```

Comment:
@kiorky, I encountered a KeyError here, the value of `ret` prior to falling into the comprehension is:

```python
[OrderedDict([('lxc.network.type', 'veth')]),
 OrderedDict([('lxc.network.name', 'eth0')]),
 OrderedDict([('lxc.network.flags', 'up')]),
 OrderedDict([('lxc.network.link', 'lxcbr0')]),
 OrderedDict([('lxc.network.hwaddr', 'AC:DE:48:0D:88:79')])]
```
I think the comprehension be `True not in ['ipv4.gateway' in a for a in ret]`